pyogrio/_io.pyx

```diff
@@ -447,20 +447,20 @@ cdef validate_feature_range(OGRLayerH ogr_layer, int skip_features=0, int max_fe
         # the count comes back as -1 if the where clause above is invalid but not rejected as error
         name = OGR_L_GetName(ogr_layer)
         warnings.warn(f"Layer '{name}' does not have any features to read")
-        feature_count = 0
-        skip_features = 0
-        max_features = 0
+        return 0, 0
 
-    else:
-        # validate skip_features, max_features
-        if skip_features < 0 or skip_features >= feature_count:
-            raise ValueError(f"'skip_features' must be between 0 and {feature_count-1}")
+    # validate skip_features, max_features
+    if skip_features < 0 or skip_features >= feature_count:
+        raise ValueError(f"'skip_features' must be between 0 and {feature_count-1}")
+
+    if max_features < 0:
+        raise ValueError("'max_features' must be >= 0")
 
-        if max_features < 0:
-            raise ValueError("'max_features' must be >= 0")
+    elif max_features == 0:
+        max_features = feature_count - skip_features
 
-        if max_features > feature_count:
-            max_features = feature_count
+    elif max_features > feature_count:
+        max_features = feature_count
 
     return skip_features, max_features
```

Comment:
One small comment: should we maybe rename this processed `max_features` (not the original keyword) to `num_features`, to make the actual meaning clearer?

Reply:
Yes, good idea.  Coming up... (sorry, should have requested review on this)

Reply:
> sorry, should have requested review on this

no, it has been here for a long time, no problem to merge (it's easy to do follow-ups if needed)